plugins/inputs/netscaler/netscaler.go

```diff
@@ -0,0 +1,194 @@
+package netscaler
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"reflect"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+/*
+ * Configuration sample for this plugin.
+ */
+var sampleConfig = `
+  ## IP address of netscaler LB
+  # host = "127.0.0.1"
+  ## Username
+  # username = "admin"
+  ## Password
+  # password = "admin"
+  ## Use https for connection
+  # https = false
+  ## Path of the Nitro API "stat" endpoint
+  # ApiPath = "/nitro/v1/stat"
+`
+
+var tr = &http.Transport{
+	TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+}
+
+var httpClient = &http.Client{
+	Timeout:   time.Second * 3,
+	Transport: tr,
+}
+
+/*
+ * This interface is satisfied by every struct resulting from a
+ * request to Netscaler's API.
+ */
+type NSResponse interface {
+	PublishMetrics(acc telegraf.Accumulator)
+}
+
+/*
+ * Main struct used by Telegraf for this plugin.
+ */
+type Netscaler struct {
+	Host     string
+	Username string
+	Password string
+	Https    bool
+	ApiPath  string
+}
+
+func (n *Netscaler) Description() string {
+	return "Telegraf plugin to gather metrics from netscaler load balancers"
+}
+
+func (n *Netscaler) SampleConfig() string {
+	return sampleConfig
+}
+
+/*
+ * Connects to a Netscaler and fetches the data.
+ * It receives the path to the resource, constructs the URL and
+ * does the connection.
+ */
+func (n *Netscaler) fetchResource(url_path string) ([]byte, error) {
+	var http_prefix string
+	var body []byte
+	var err error = nil
+
+	if n.Https {
+		http_prefix = "https://"
+	} else {
+		http_prefix = "http://"
+	}
+	url := http_prefix + n.Host + n.ApiPath + url_path
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return body, err
+	}
+	// Headers required by Netscaler's API interface
+	req.Header.Add("X-NITRO-USER", n.Username)
+	req.Header.Add("X-NITRO-PASS", n.Password)
+	req.Header.Add("Content-type", "application/x-www-form-urlencoded")
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		log.Fatal("Could not connect to the host:", err)
+		return body, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		return body, err
+	}
+
+	body, err = ioutil.ReadAll(resp.Body)
+	return body, err
+}
+
+// Dispatches metrics to Telegraf
+func (n *Netscaler) Gather(acc telegraf.Accumulator) error {
+	resources := make(map[string]interface {
+		PublishMetrics(acc telegraf.Accumulator)
+	})
+	resources["/lbvserver/"] = new(LBVResponse)
```

Comment:
I don't really get this whole part, is this for future proof? Because you are only getting this 1 resource only right? Also new() returns a pointer, then you are passing the reference of the value to Unmarshal with &resStruct, isn't that a pointer to a pointer?

Reply:
Somehow it worked, but you are right.
Removed unnecessary "&".